### PR TITLE
[WFCORE-1863] Make CredentialReference thread safe

### PR DESCRIPTION
**Pull request by pskopek**
making CredentialReference thread safe as per comment in https://github.com/wildfly/wildfly-core/pull/1816
https://issues.jboss.org/browse/WFCORE-1863

### CODE REVIEW

File: controller/src/main/java/org/jboss/as/controller/security/CredentialReference.java [hunk 1]
```diff
@@ -164,7 +163,6 @@ public void destroy() {
             for (int i = 0; i < secret.length; i++) {
                 secret[i] = 0;
             }
-            secret = null;
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/security/CredentialReference.java]: This breaks isDestroyed().

Comment (pskopek) [inline controller/src/main/java/org/jboss/as/controller/security/CredentialReference.java]: Thanks for spotting it @bstansberry !

**Comment by pskopek** [comment]
New changes pushed.

**Comment by dmlloyd** [comment]
Actually simply changing the "secret" field to be volatile should be sufficient to make this class thread-safe.  All the other stuff just adds bulk and complexity that is unwarranted AFAICT.

**Comment by bstansberry** [comment]
@pskopek any thoughts re: dmlloyd's comment?

**Comment by pskopek** [comment]
I have made the changes dmlloyd suggested.

**Comment by dmlloyd** [comment]
Looks fine to me now.

**Pull request merged**